### PR TITLE
Make CLI Fail Gracefully on Dep Script Errors

### LINKED ISSUE #8471: 0.65+: run-windows will error out if multiple VS installed
Filing on behalf of @tido64 who reached out with this issue

In 0.65 we added running the rnw-depependencies script to CLI health checks. https://github.com/microsoft/react-native-windows/pull/7090

This means any rnw-dependencies bugs will block the build from starting. A bug was identified, where multiple versions of Visual Studio is installed.

MicrosoftTeams-image (7)

The bug that causes this is present in several places in the dependencies script. We execute `vswhere`, which returns multiple paths, but concatenate assuming that only a single path is returned. This creates an invalid path to search for.

```
function CheckCppWinRT_VSIX {
    try {
        $vsWhere = "${env:ProgramFiles(x86)}\Microsoft Visual Studio\Installer\vswhere.exe"
        if (!(Test-Path $vsWhere)) {
            return $false;
        }
        $vsPath = & $vsWhere -version 16.5 -property installationPath;
        $natvis = Get-ChildItem "$vsPath\Common7\IDE\Extensions\cppwinrt.natvis" -Recurse;
        return $null -ne $natvis;
    } catch { return $false };
}
```

### PR DESCRIPTION
**Pull request by NickGerleman**
Fixes #8471

Should potentially be backported to 0.65/0.66 if we see more folks running into issues here.

### CODE REVIEW

**Comment by NickGerleman** [comment]
cc @asklar who wrote the deps script.

I've been back and forward on this change. This is muting signal that can be used to diagnose issues in the deps script, but I think there have been other corner cases in the deps script we have been okay with. Inability to build is a pretty bad end result for customers though, so my inclination would be to set a boundary around the damage if we know there are cases where things may go wrong.

**Comment by NickGerleman** [comment]
Ideally we could log some telemetry here, without throwing an unhandled exception. cc @jonthysell who has been working on CLI telemetry in case he has thoughts.

**Review by dannyvv — APPROVED** [review]

**Comment by acoates-ms** [comment]
Could we maybe have it at least return a health check that when run says that it failed to get the health checks?  Then at least `react-native doctor` calls wouldn't silently succeed when there is an issue?

File: packages/@react-native-windows/cli/src/healthChecks.ts [hunk 1]
```diff
@@ -19,7 +36,7 @@ export function getHealthChecks(): HealthCheckCategory[] | undefined {
         'react-native-windows/package.json',
         {paths: [process.cwd()]})), 'Scripts/rnw-dependencies.ps1');
     
-    const rnwDeps = execSync(`powershell -ExecutionPolicy Unrestricted -NoProfile ${rnwDepScriptPath} -NoPrompt -ListChecks`);
+    const rnwDeps = execSync(`powershell -ExecutionPolicy Unrestricted -NoProfile ${rnwDepScriptPath} -NoPrompt -ListChecks`, {stdio: 'pipe'});
```
Comment (acoates-ms) [inline packages/@react-native-windows/cli/src/healthChecks.ts:39]: Does the pipe here end up spewing a bunch of output into the normal `react-native doctor` command?  I understand you are probably trying to gain some useful output for cases where the script fails.  Just want to understand if we are making a trade off of polluting the normal output?

Comment (NickGerleman) [reply packages/@react-native-windows/cli/src/healthChecks.ts:39]: Opposite. By default execSync will output `stderr` to the parent process output. `inherit` directs it just to the subprocess stdout, stderr, etc.

Comment (NickGerleman) [reply packages/@react-native-windows/cli/src/healthChecks.ts:39]: We previously got this gnarly output. 
image

Comment (NickGerleman) [reply packages/@react-native-windows/cli/src/healthChecks.ts:39]: After change: 
image

Comment (NickGerleman) [reply packages/@react-native-windows/cli/src/healthChecks.ts:39]: Verified the correct case still works.
image

Comment (NickGerleman) [reply packages/@react-native-windows/cli/src/healthChecks.ts]: This falls further into the "eating the error" territory, but folks can always run the deps script manually to see the full output of why it is failing.

**Pull request merged**